### PR TITLE
[Student][Builds][MBL-13398]: Track student build stats in Sumologic

### PR DESCRIPTION
**Pull request by joehoag**
For now, we just report build data for successful Bitrise runs.

UPDATE:  I also stubbed out a couple of tests that won't pass consistently until we fix MBL-13405.

### CODE REVIEW

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1008762&projectid=20)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

File: apps/student/build.gradle [hunk 1]
```diff
@@ -281,4 +283,131 @@ dependencies {
     implementation Libs.NEW_RELIC
 }
 
+// A listener that will allow us to monitor task timings and, eventually, APK size
+class TimingsListener implements TaskExecutionListener, BuildListener {
```
Comment (CalvinKern) [inline apps/student/build.gradle]: Is there any way to declare this class in a standalone file, so we can reuse this for teacher/parent if we want to? Not very familiar with gradle scripts like this, so I'm not sure if this is possible.

Comment (joehoag) [reply apps/student/build.gradle]: That's a good idea.  I was going to tackle that issue when I extended the same logic to teacher and (maybe) parent.

File: apps/student/build.gradle [hunk 2]
```diff
@@ -281,4 +283,131 @@ dependencies {
...
+
+    @Override
+    void afterExecute(Task task, TaskState taskState) {
+        def ms = TimeUnit.MILLISECONDS.convert(System.nanoTime() - startTime, TimeUnit.NANOSECONDS)
+        timings.put(task.name, ms)
+
+        //task.project.logger.warn "${task.path} took ${ms}ms"
+        if (task.name.contains("assemble")) {
+            sawAssembleTask = true
+        }
```
Comment (CalvinKern) [inline apps/student/build.gradle]: Not sure what this `sawAssembleTask` flag is used for (dead variable?) but may be worth an OR check if the name contains "bundle" for when we switch over to bundled releases (though we will only generate bundles for release builds, not sure if this needs to consider that).

Comment (joehoag) [reply apps/student/build.gradle]: Right now it's not used at all.  I toyed around with the idea of only reporting information if a build occurred.  Later I pivoted to just reporting from successful bitrise builds.  I can take this out.

Comment (joehoag) [reply apps/student/build.gradle]: But I think you're write that I might have to change my apk-finding logic after your aab changes are merged in.

File: apps/student/build.gradle (hunk 1 above)
Comment (CalvinKern) [reply apps/student/build.gradle]: Cool. If it works now, then we can leave it like this and tackle migrating when we include other apps.

File: apps/student/build.gradle (hunk 2 above)
Comment (CalvinKern) [reply apps/student/build.gradle]: If it's doing this on bitrise, we will also be building a universal apk (for ease of testing). And that's probably more important to note for size than the resulting aab (as it still needs to get converted into apks and isn't actually the size we install for users).

File: apps/student/build.gradle (hunk 1 above)
Comment (joehoag) [reply apps/student/build.gradle]: Working on a solution...

**Comment by CalvinKern** [comment]
Looks good to me! Nice work getting that class separated out!

**Comment by joehoag** [comment]
Thanks for the review!

**Pull request merged**